### PR TITLE
feat: Rocm dockerfile

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Description

Add PyTorch ROCm hpc and non-hpc image to repo for users to build.

## Checklist

- [x] Bump VERSION to make the pushed images are tagged with the right version.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] Test the images by running the test `bumpenvs` procedure in the determined repo. See [README](https://github.com/determined-ai/environments/blob/main/README.md).

### CODE REVIEW

File: Dockerfile-infinityhub-hpc [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+RUN apt install rocm-libs
+RUN apt remove -y openmpi ucx
+#Let's remove existing /opt/ompi; and, link to our version.
+RUN rm -rf /opt/ompi
+RUN ln -s /container/ompi /opt
+COPY dockerfile_scripts /tmp/det_dockerfile_scripts
+
+# SHOULDN'T NEED TO SET SOME OF THESE VARIABLES
+#USING OFI
```
Comment (jgongd) [inline Dockerfile-infinityhub-hpc:13]: > Shouldn't need to set some of these variables 
Remove the ones that are not required?

File: Makefile [hunk 2]
```diff
@@ -140,6 +145,15 @@ NGC_PYTORCH_HPC_REPO := pytorch-ngc-hpc-dev
 NGC_TF_REPO := tensorflow-ngc-dev
 NGC_TF_HPC_REPO := tensorflow-ngc-hpc-dev
 
+INFINITYHUB_PYTORCH_PREFIX := rocm/pytorch
+INFINITYHUB_TENSORFLOW_PREFIX := rocm/tensorflow
+INFINITYHUB_PYTORCH_VERSION := 2.1.2
+INFINITYHUB_TENSORFLOW_VERSION :=
```
Comment (jgongd) [inline Makefile:151]: Are these used somewhere?

File: Makefile [hunk 3]
```diff
@@ -140,6 +145,15 @@ NGC_PYTORCH_HPC_REPO := pytorch-ngc-hpc-dev
...
 NGC_TF_HPC_REPO := tensorflow-ngc-hpc-dev
 
+INFINITYHUB_PYTORCH_PREFIX := rocm/pytorch
+INFINITYHUB_TENSORFLOW_PREFIX := rocm/tensorflow
+INFINITYHUB_PYTORCH_VERSION := 2.1.2
+INFINITYHUB_TENSORFLOW_VERSION := 
+export INFINITYHUB_PYTORCH_REPO := pytorch-infinityhub-dev
+INFINITYHUB_PYTORCH_HPC_REPO := pytorch-infinityhub-hpc-dev
+INFINITYHUB_TF_REPO := tensorflow-infinityhub-dev
+INFINITYHUB_TF_HPC_REPO := tensorflow-infinityhub-hpc-dev
```
Comment (jgongd) [inline Makefile:155]: Is TF ROCm images out of the scope of this PR?

File: Dockerfile-infinityhub-hpc (hunk 1 above)
Comment (MikhailKardash) [reply Dockerfile-infinityhub-hpc]: Removed this comment

File: Makefile (hunk 2 above)
Comment (MikhailKardash) [reply Makefile]: Removed TF references

File: Makefile (hunk 3 above)
Comment (MikhailKardash) [reply Makefile]: Removed TF references

File: Makefile [hunk 4]
```diff
@@ -163,39 +174,151 @@ build-tensorflow-ngc:
...
+ROCM61_TORCH_MPI :=pytorch-2.0-tf-2.10-rocm-mpich
 else
-ROCM56_TORCH_MPI :=pytorch-2.0-tf-2.10-rocm-ompi
+ROCM61_TORCH_MPI :=pytorch-2.0-tf-2.10-rocm-ompi
 endif
-export ROCM56_TORCH_TF_ENVIRONMENT_NAME := $(ROCM_56_PREFIX)$(ROCM56_TORCH_MPI)
-.PHONY: build-pytorch20-tf210-rocm56
-build-pytorch20-tf210-rocm56:
+
+export ROCM61_TORCH_TF_ENVIRONMENT_NAME := $(ROCM_60_PREFIX)$(ROCM61_TORCH_MPI)
```
Comment (jgongd) [inline Makefile:220]:
```suggestion
export ROCM61_TORCH_TF_ENVIRONMENT_NAME := $(ROCM_61_PREFIX)$(ROCM61_TORCH_MPI)
```

File: Makefile [hunk 5]
```diff
@@ -163,39 +174,151 @@ build-tensorflow-ngc:
...
-ROCM56_TORCH13_MPI :=pytorch-1.3-tf-2.10-rocm-mpich
+ROCM61_TORCH13_MPI :=pytorch-1.3-tf-2.10-rocm-mpich
 else
-ROCM56_TORCH13_MPI :=pytorch-1.3-tf-2.10-rocm-ompi
+ROCM61_TORCH13_MPI :=pytorch-1.3-tf-2.10-rocm-ompi
 endif
-export ROCM56_TORCH13_TF_ENVIRONMENT_NAME := $(ROCM_56_PREFIX)$(ROCM56_TORCH13_MPI)
-.PHONY: build-pytorch13-tf210-rocm56
-build-pytorch13-tf210-rocm56:
+export ROCM61_TORCH13_TF_ENVIRONMENT_NAME := $(ROCM_60_PREFIX)$(ROCM61_TORCH13_MPI)
```
Comment (jgongd) [inline Makefile:200]:
```suggestion
export ROCM61_TORCH13_TF_ENVIRONMENT_NAME := $(ROCM_61_PREFIX)$(ROCM61_TORCH13_MPI)
```

File: Makefile [hunk 6]
```diff
@@ -163,39 +174,151 @@ build-tensorflow-ngc:
...
+.PHONY: build-pytorch13-tf210-rocm60
+build-pytorch13-tf210-rocm60:
```
Comment (jgongd) [inline Makefile:202]: I'm a bit confused. The base image is on rocm6.0 and pytorch-1.3-tf-2.10-rocm-mpich/ompi is on rocm6.1?

File: Makefile [hunk 7]
```diff
@@ -140,6 +145,12 @@ NGC_PYTORCH_HPC_REPO := pytorch-ngc-hpc-dev
 NGC_TF_REPO := tensorflow-ngc-dev
 NGC_TF_HPC_REPO := tensorflow-ngc-hpc-dev
 
+INFINITYHUB_PYTORCH_PREFIX := rocm/pytorch
+INFINITYHUB_TENSORFLOW_PREFIX := rocm/tensorflow
+INFINITYHUB_PYTORCH_VERSION := 2.1.2
```
Comment (jgongd) [inline Makefile:150]: It looks like these are not used.

Comment (jgongd) [reply Makefile:150]: Oh, is `INFINITYHUB_PYTORCH_PREFIX` intended to be used at line 184: BASE_IMAGE="rocm/pytorch:rocm6.1_ubuntu22.04_py3.10_pytorch_2.1.2" \
                 --build-arg

File: Makefile [hunk 8]
```diff
@@ -163,39 +174,151 @@ build-tensorflow-ngc:
...
-build-pytorch20-tf210-rocm56:
+
+export ROCM61_TORCH_TF_ENVIRONMENT_NAME := $(ROCM_60_PREFIX)$(ROCM61_TORCH_MPI)
+.PHONY: build-pytorch20-tf210-rocm60
+build-pytorch20-tf210-rocm60:
 	docker build -f Dockerfile-default-rocm \
-		--build-arg BASE_IMAGE="rocm/pytorch:rocm5.6_ubuntu20.04_py3.8_pytorch_2.0.1" \
-		--build-arg TENSORFLOW_PIP="tensorflow-rocm==2.10.1.540" \
-		--build-arg HOROVOD_PIP="horovod==0.28.1" \
+                --build-arg BASE_IMAGE="rocm/pytorch:rocm6.1_ubuntu22.04_py3.10_pytorch_2.1.2" \
```
Comment (jgongd) [inline Makefile:224]: Should the base_image be rocm6.0 since this is `build-pytorch20-tf210-rocm60`?

File: Makefile (hunk 6 above)
Comment (MikhailKardash) [reply Makefile]: Bumped these all to rocm6.1

File: Makefile (hunk 7 above)
Comment (MikhailKardash) [reply Makefile]: Removed these

File: Makefile (hunk 8 above)
Comment (MikhailKardash) [reply Makefile:221]: bumped to rocm 6.1

File: Makefile [hunk 9]
```diff
@@ -163,39 +171,151 @@ build-tensorflow-ngc:
...
+.PHONY: build-pytorch20-tf210-rocm61
+build-pytorch20-tf210-rocm61:
+	docker build -f Dockerfile-default-rocm \
+                --build-arg BASE_IMAGE="rocm/pytorch:rocm6.1_ubuntu22.04_py3.10_pytorch_2.1.2" \
+                --build-arg TENSORFLOW_PIP="tensorflow-rocm==2.10.1.540" \
+                --build-arg HOROVOD_PIP="0" \
+                --build-arg WITH_MPICH=$(WITH_MPICH) \
+                -t $(DOCKERHUB_REGISTRY)/$(ROCM61_TORCH_TF_ENVIRONMENT_NAME)-$(SHORT_GIT_HASH) \
+                -t $(DOCKERHUB_REGISTRY)/$(ROCM61_TORCH_TF_ENVIRONMENT_NAME)-$(VERSION) \
+                .
```
Comment (jgongd) [inline Makefile:246]: We have two `build-pytorch20-tf210-rocm61`.

**Review by jgongd — APPROVED** [review]
Just one more comment. Everything else LGTM!

File: Makefile (hunk 9 above)
Comment (MikhailKardash) [reply Makefile]: Removed the duplicate

**Review by will-HPE — APPROVED** [review]

**Pull request merged**